yb-voyager/cmd/importData.go

```diff
@@ -652,6 +664,25 @@ func importData(importFileTasks []*ImportFileTask) {
 
 }
 
+func startAdaptiveParallelism() (bool, error) {
+	yb, ok := tdb.(*tgtdb.TargetYugabyteDB)
+	if !ok {
+		return false, fmt.Errorf("adaptive parallelism is only supported if target DB is YugabyteDB")
+	}
+	if !yb.IsAdaptiveParallelismSupported() {
+		utils.PrintAndLog(color.YellowString("Note: Continuing without adaptive parallelism as it is not supported in this version of YugabyteDB."))
+		return false, nil
+	}
+
+	go func() {
+		err := adaptiveparallelism.AdaptParallelism(yb)
+		if err != nil {
+			log.Errorf("adaptive parallelism error: %v", err)
+		}
```

Comment:
Just thinking, shall we take some if err is not null for more than x number of times(let's say)
Similar comment about the err condition inside `adaptiveparallelism.AdaptParallelism()`

Reply:
I agree with the general idea, but this needs more thought. It'll probably be a bad idea to do this for all types of errors.